recipes/yaml/CMakeLists.txt.patch

```diff
@@ -0,0 +1,25 @@
+diff --git CMakeLists.txt CMakeLists.txt
+index e84c28c..eab70b2 100644
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -3,6 +3,13 @@
+ cmake_minimum_required (VERSION 2.8)
+ project (yaml C)
+ 
++# Allow the developer to select if Dynamic or Static libraries are built
++OPTION (BUILD_SHARED_LIBS "Build Shared Libraries" OFF)
++SET (LIB_TYPE STATIC)
++IF (BUILD_SHARED_LIBS)
++    SET (LIB_TYPE SHARED)
++ENDIF (BUILD_SHARED_LIBS)
++
+ set (YAML_VERSION_MAJOR 0)
+ set (YAML_VERSION_MINOR 1)
+ set (YAML_VERSION_PATCH 6)
+@@ -12,5 +19,4 @@ file (GLOB SRC src/*.c)
+ 
+ include_directories (include win32)
+ add_definitions (-DHAVE_CONFIG_H -DYAML_DECLARE_STATIC)
+-add_library (yaml STATIC ${SRC})
+-
++add_library (yaml ${LIB_TYPE} ${SRC})
```

Comment:
Going by the CMake file by @gillins on the jpeg recipe, I think you can achieve the same effect by [adding two libraries](https://github.com/conda-forge/staged-recipes/pull/355/files#diff-e19d5d6f96ff83dbcda9dbd79d2b0a36R23).

Reply:
Sorry to be necromancing an ancient topic, but this static lib name change did indeed break a number of things (as some of the downstream consumers of libyaml are very stable and not often updated, it's taken awhile to discover the issues). We've changed the default back, but made it configurable from cmake (see https://github.com/yaml/libyaml/pull/136) so anyone that might need the new name can at least get it back if they want.